### ISSUE TITLE
`--no-check` renames classes and shouldn't

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#11811]
Ref: https://github.com/swc-project/swc/issues/2137

Given the code:

**example.ts**

```ts
export class Response {
  log() {
    console.log(this.constructor.name);
  }
}
```

And then `deno run --no-check example.ts` will emit the following:

```js
class Response1 {
    log() {
        console.log(this.constructor.name);
    }
}
export { Response1 as Response };
```

Thereby causing a change in value of `this.constructor.name`.  I have found also a class name `Request` causes the same behaviour.

**Comment by kitsonk** [comment]
As mentioned by @kdy1 when dealing with the upstream issue there are additional passes we will need to leverage:

> Also, note that you need to use `unique_scope` and `hygiene_optimizer` from `swc_ecma_minifer` to use the result of the PR.

**Comment by kdy1** [comment]
Currently, minifier is not reexported from `swc_ecmascript`, but https://github.com/swc-project/swc/pull/2146 will fix it.

**Comment by kitsonk** [comment]
Subsequently https://github.com/swc-project/swc/pull/2193 was added which meant that the hygiene pass now is sufficient and I have checked that updating swc has resolved this issue.  It is currently working in canary and will be released in Deno 1.14.
